src/sidetree/index.ts

```diff
@@ -0,0 +1,209 @@
+import { randomBytes } from 'react-native-randombytes';
+import { Ed25519KeyPair } from '@transmute/did-key-ed25519';
+import { Secp256k1KeyPair } from '@transmute/did-key-secp256k1'
+import { IonRequest, LocalSigner } from '@decentralized-identity/ion-sdk';
+
+
+import { saveItem, getItem } from '../store/';
+import {logger} from "../logging";
+
+//TODO UPDATE, RECOVERY
+
+//TODO MOVE URL TO CONFIG
+//const nodeURL = 'https://testnet.sidetree-cardano.com/cardano';
+const nodeURL = 'http://192.168.86.38:3000';
+
+
+export async function generateKeyPair(type: string) {
+    try {
+        let keyGenerator = Secp256k1KeyPair;
+        if (type === 'Ed25519') { keyGenerator = Ed25519KeyPair; };
+        const keyPair = await keyGenerator.generate({
+            secureRandom: () => randomBytes(32)
+        });
+        const { publicKeyJwk, privateKeyJwk } = await keyPair.export({
+            type: 'JsonWebKey2020',
+            privateKey: true,
+        });
+        return {
+            publicJwk: publicKeyJwk,
+            privateJwk: privateKeyJwk
+        };
+    } catch (error: any) {
+        logger("sidetree - Error", error)
```

Comment:
the logger truncates large output.  but for errors that isn't desired... use:
console.error("sidetree - Error",error,error.stack) so that we can see full stack traces
there are 7 of these to change.